warden/src/main/kotlin/AttestationService.kt

```diff
@@ -324,6 +291,35 @@ abstract class AttestationService {
         }
     }
 
+    private fun <T : PublicKey> processAndroidAttestationResult(
+        keyToBeAttested: T,
+        firstTry: AttestationResult.Android
+    ): KeyAttestation<T> =
+        if (CryptoPublicKey.fromJcaPublicKey(keyToBeAttested) == CryptoPublicKey.fromJcaPublicKey(
+                firstTry.attestationCertificate.publicKey
+            )
+        ) KeyAttestation(keyToBeAttested, firstTry)
+        else {
+            ("Android attestation failed: keyToBeAttested (${keyToBeAttested.encoded.encodeBase64()}) does not match " +
+                    "key from attestation certificate: ${firstTry.attestationCertificate.publicKey.encoded.encodeBase64()}").let {
+                KeyAttestation(
+                    null,
+                    AttestationResult.Error(
+                        explanation = it,
+                        cause = AttException.Content.Android(
+                            it,
+                            AttestationValueException(
+                                it,
```

Comment:
Much better, but this still seems off, because not every parameter is using the parameter name, and it takes up 14 lines to create the result